src/azure-cli/azure/cli/command_modules/eventgrid/_params.py

```diff
@@ -61,29 +172,81 @@ def load_arguments(self, _):    # pylint: disable=too-many-statements
+        c.argument('delivery_identity_endpoint_type', arg_type=get_enum_type(['webhook', 'eventhub', 'storagequeue', 'hybridconnection', 'servicebusqueue', 'servicebustopic', 'azurefunction'], default=None), is_preview=True)
         c.argument('source_resource_id', help="Fully qualified identifier of the source Azure resource.")
-        c.argument('resource_id', deprecate_info=c.deprecate(redirect="--source-resource-id", expiration='3.0.0', hide=True), help="Fully qualified identifier of the Azure resource.")
-        c.argument('endpoint', help="Endpoint where EventGrid should deliver events matching this event subscription. For webhook endpoint type, this should be the corresponding webhook URL. For other endpoint types, this should be the Azure resource identifier of the endpoint. It is expected that the destination endpoint to be already created and available for use before executing any Event Grid command.")
+        c.argument('endpoint', help="Endpoint where EventGrid should deliver events matching this event subscription. For webhook endpoint type, this should be the corresponding webhook URL. For other endpoint types, this should be the Azure resource identifier of the endpoint.")
+        c.argument('delivery_identity_endpoint', help="Endpoint with identity where EventGrid should deliver events matching this event subscription. For webhook endpoint type, this should be the corresponding webhook URL. For other endpoint types, this should be the Azure resource identifier of the endpoint.", is_preview=True)
         c.argument('event_subscription_name', help="Name of the event subscription.")
         c.argument('subject_begins_with', help="An optional string to filter events for an event subscription based on a prefix. Wildcard characters are not supported.")
         c.argument('subject_ends_with', help="An optional string to filter events for an event subscription based on a suffix. Wildcard characters are not supported.")
         c.argument('topic_type_name', help="Name of the topic type.")
         c.argument('is_subject_case_sensitive', arg_type=get_three_state_flag(), options_list=['--subject-case-sensitive'], help="Specify to indicate whether the subject fields should be compared in a case sensitive manner. True if flag present.", )
+        c.argument('input_mapping_fields', arg_type=input_mapping_fields_type)
+        c.argument('input_mapping_default_values', arg_type=input_mapping_default_values_type)
+        c.argument('input_schema', arg_type=input_schema_type)
         c.argument('odata_query', arg_type=odata_query_type)
         c.argument('domain_name', arg_type=domain_name_type)
         c.argument('domain_topic_name', arg_type=domain_topic_name_type)
+        c.argument('system_topic_name', arg_type=system_topic_name_type)
+        c.argument('source', help="The ARM Id for the topic, e.g., /subscriptions/{SubId}/resourceGroups/{RgName}/providers/Microsoft.Storage/storageAccounts/{AccountName}")
```

Comment:
--source should be arm resource id for topic, right? could we use topic resource id as example not storage account resource id?

Reply:
No. This is not same as resource id. the word "source" needs to be maintained and used here. Also, this copied exactly from Event Grid CLI extension and user are using it this way.

Reply:
From the example you provided, it is storage account resource id, right? CLI extension is used for private preview commands and for cli modules, we need to be consistent with cli command guidline.
here we want the description for --source could be more clear. If it is not resource id, could you provide another example?

Reply:
@Juliehzl source is a concept used in Event Grid and messaging world. it just happens to be the arm id for the system topic (Storage in this case). We will not change the name (source) as this is the expected name we want to use in our documentation and command. this is unique to event grid and we use source-resource-id in event subsription creation/update command and yes it matches what you are talking about. The case here is totally different.

This is still by design and won't fix.